src/dcap_provider.cpp

```diff
@@ -897,52 +1009,138 @@ extern "C" quote3_error_t sgx_ql_get_quote_config(
 
     try
     {
-        const std::string cert_url = build_pck_cert_url(*p_pck_cert_id);
-        if (auto cache_hit = try_cache_get(cert_url))
+        collateral_fetch_url collateral_url;
+        std::vector<uint8_t> response_body;
+        std::string temp;
+        build_pck_cert_url(*p_pck_cert_id, collateral_url);
+        std::string primary_base_url = collateral_url.primary_base_url.str();
+        std::string secondary_base_url =
+            collateral_url.secondary_base_url.str();
+        std::unique_ptr<curl_easy> curl;
+        bool fetch_response_from_primary = false;
+        sgx_ql_config_t temp_config{};
+        nlohmann::json json_body;
+        std::string fetch_from_base = fetch_from_base_url();
+        transform(
+            fetch_from_base.begin(),
+            fetch_from_base.end(),
+            fetch_from_base.begin(),
+            ::tolower);
+        if (fetch_from_base.compare("true") == 0)
         {
-            log(SGX_QL_LOG_INFO,
-                "Fetching quote config from cache: '%s'.",
-                cert_url.c_str());
+            try
+            {
+                curl = curl_easy::create(primary_base_url, nullptr, 0);
+                log(SGX_QL_LOG_INFO,
+                    "Fetching quote config from host agent: '%s'.",
+                    primary_base_url.c_str());
+                curl->set_headers(headers::localhost_metadata);
+                curl->perform();
+                fetch_response_from_primary = true;

```

Comment:
This should get set when JSON has been parsed. nit: variable name could indicate that primary_Success